### PR TITLE
Add updatevault dftx

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Add updatevault dftx

#### Which issue(s) does this PR fixes?:

Fixes part of #532

### CODE REVIEW

File: packages/jellyfish-transaction/__tests__/script/dftx/dftx_loans/UpdateVault.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import {
+  CUpdateVault,
+  UpdateVault
+} from '../../../../src/script/dftx/dftx_loans'
+import { OP_CODES } from '../../../../src/script'
+import { toBuffer, toOPCodes } from '../../../../src/script/_buffer'
+import { OP_DEFI_TX } from '../../../../src/script/dftx'
+
+it('should bi-directional buffer-object-buffer', () => {
+  const fixtures = [
```
Comment (canonbrother) [inline packages/jellyfish-transaction/__tests__/script/dftx/dftx_loans/UpdateVault.test.ts:11]: feels like same conditions.. are these included update - only ownerAddr and only scheme??

**Review by canonbrother — COMMENTED** [review]
left a question
and overall good looks good for me.. although that has some missing test on txn builder but those is actually covered by rpc (the validations are same path)

File: packages/jellyfish-transaction/__tests__/script/dftx/dftx_loans/UpdateVault.test.ts (hunk 1 above)
Comment (jingyi2811) [reply packages/jellyfish-transaction/__tests__/script/dftx/dftx_loans/UpdateVault.test.ts:11]: Ya, I left it intentionally. 
This is what I think. For DFTXs, all the parameters must exists.

Comment (jingyi2811) [reply packages/jellyfish-transaction/__tests__/script/dftx/dftx_loans/UpdateVault.test.ts:11]: And also, like the UTXOS tests in the RPC. I won't add them too.

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_update_vault.test.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    await alice.generate(1)
+
+    // Take loan
+    await alice.rpc.loan.takeLoan({
+      vaultId,
+      amounts: '50@TSLA'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '0.9@DFI', currency: 'USD' }] })
```
Comment (surangap) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_update_vault.test.ts:163]: generate a block after this? coz of https://github.com/DeFiCh/jellyfish/pull/816 ?
```suggestion
    await alice.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '0.9@DFI', currency: 'USD' }] })
    await alice.generate(1)
```

Comment (jingyi2811) [reply packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_update_vault.test.ts:163]: Will do. You are too fast.

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**